src/Parsable.php

```diff
@@ -21,40 +21,39 @@ abstract protected function useAssignmentIfEmpty(): bool;
 
     public function __construct(string|array $parsable = null)
     {
-        if ($parsable !== null) {
-            if (is_string($parsable)) {
-                if (substr($parsable, 0, 1) == $this->getParsedSeperator()) {
-                    $parsable = substr($parsable, 1);
-                }
-                //manually parse to check key names for control chars later
-                $rawKeys = [];
-                foreach (explode('&', $parsable) as $pair) {
-                    $exploded = explode('=', $pair);
-                    $rawKeys[$exploded[0]] = true;
-                }
-                parse_str($parsable, $pairs);
-
-                // check and repair key names with periods
-                $finalParams = [];
-                foreach ($pairs as $key => $value) {
-                    $possibleRepairedKey = str_replace('_', '.', $key);
-                    if (isset($rawKeys[$possibleRepairedKey])) {
-                        $key = $possibleRepairedKey;
-                    }
-                    $finalParams[$key] = $value;
-                }
-                $pairs = $finalParams;
-            } else {
-                $pairs = $parsable;
+        if ($parsable === null) {
+            return;
+        }
+
+        if (is_string($parsable)) {
+            if (substr($parsable, 0, 1) == $this->getParsedSeperator()) {
+                $parsable = substr($parsable, 1);
             }
 
-            if (! is_array($pairs)) {
-                throw new InvalidArgumentException('$pairs must be an array or a query string');
```

Comment:
This cannot happen.
`null` is handled, `string` is handled, passing in something other than an array causes a fatal error.

Reply:
Note that without `declare(strict_types = 1);` e.g integers will be type-juggled to strings.